examples/kubernetes/sc-allowedtopo.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: csi-filestore-allowedtopo
+provisioner: filestore.csi.storage.gke.io
+volumeBindingMode: WaitForFirstConsumer
+allowVolumeExpansion: true
+allowedTopologies:
```

Comment:
Do you see a need for a use case where someone wants to provision an instance in a different zone as the pod?

Currently the scheduling logic will cause the Pod to be scheduled in the [same zones](https://github.com/kubernetes/kubernetes/blob/3647766cbcb39558291568b5cd0e38ad34a9feeb/pkg/controller/volume/scheduling/scheduler_binder.go#L865) during the first provisioning.

Reply:
I am not sure of such a case. However, allowedTopologies will allow user to restrict the candidate zones where a FS instance can be deployed, which this example highlights. say for an regional cluster, user can provide a subset of zones where the FS instance can be deployed.(although it will be still accesible to the entities in the same VPC)

Reply:
Wouldn't the flow in this case be to create a PV via a bindingMode=ImmediatePVC and a requisite topology specifying zone X for the filestore instance, then a pod with an affinity to a node pool in zone Y?

We could document that in the examples or docs (assuming I got it right...)

I have a hard time seeing how one could make this any simpler in practice without extending the existing topology representation (?)

Reply:
Yes, I think immediate binding and setting allowed topologies is the way to address this scenario.

Reply:
Added 1 example each for both delayed binding and immediate binding. The other scenarios are listed in the behavior matrix in README